src/battle_game/core/bullets/boomerang.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include "battle_game/core/bullet.h"
+
+namespace battle_game::bullet {
+class Boomerang : public Bullet {
+ public:
+  Boomerang(GameCore *core,
+            uint32_t id,
+            uint32_t unit_id,
+            uint32_t player_id,
+            glm::vec2 position,
+            float rotation,
+            float damage_scale,
+            glm::vec2 velocity);
+  ~Boomerang() override;
+  void Render() override;
+  void Update() override;
+
+ private:
+  glm::vec2 velocity_{};
+  int time_{0};
+  int hit_{0};
+  bool return_{false};
```

Comment:
These variables-naming is not really understandable, maybe some comments will be better.

Reply:
OK,I'll pay attention next time.
Here time_ is for how long the bullet has been flying;
hit_ is for the times the bullet hits a unit (it does not disappear after one hit but two)
return_ is for whether its flying away from the tank or returning to it.